app/scripts/controllers/user-storage/user-storage-controller.ts

```diff
@@ -111,11 +131,32 @@ export default class UserStorageController extends BaseController<
         'AuthenticationController:performSignIn',
       );
     },
+    signOut: async () => {
+      return await this.messagingSystem.call(
+        'AuthenticationController:performSignOut',
+      );
+    },
+  };
+
+  #metamaskNotifications = {
+    disableMetamaskNotifications: async () => {
+      return await this.messagingSystem.call(
+        'MetamaskNotificationsController:disableMetamaskNotifications',
```

Comment:
Personal nit (we can figure this out in future)

The messaging system makes this a web of messages, there is no control flow.

E.g.
Before the structure was below, where we kindof had a centralised place where each of the controllers talked to.

```mermaid
graph TD;
  AuthenticationController-->NotificationController;
  UserStorageController-->NotificationController;
  PushNotificationController-->NotificationController;
```

But now we have all the controllers communicating with one another... feels messy.

Future - I want to refactor this so we have a single controller that pieces together all the possible User Actions.

E.g.
- Enable Notifications
- Disable Notifications
- Fetch Notifications
- Etc.

Reply:
Ticket: https://consensyssoftware.atlassian.net/browse/NOTIFY-658